public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -958,6 +959,32 @@ export class CloudWatchDatasource extends DataSourceApi<CloudWatchQuery, CloudWa
       metricsQueries,
     };
   };
+
+  interpolateVariablesInQueries(queries: CloudWatchQuery[], scopedVars: ScopedVars): CloudWatchQuery[] {
+    if (!queries.length) {
+      return queries;
+    }
+
+    return queries.map(query => ({
+      ...query,
+      region: this.getActualRegion(this.replace(query.region, scopedVars)),
+      expression: this.replace(query.expression, scopedVars),
+
+      ...(!isCloudWatchLogsQuery(query) && this.interpolateMetricsQueryVariables(query, scopedVars)),
+    }));
+  }
+
+  interpolateMetricsQueryVariables(
+    query: CloudWatchMetricsQuery,
+    scopedVars: ScopedVars
+  ): Pick<CloudWatchMetricsQuery, 'alias' | 'metricName' | 'namespace' | 'period'> {
```

Comment:
I think that `dimensions` have to be interpolated too.
Consider a query like:
```
{
      "namespace": "$namespace",
      "metricName": "$metricName",
      "expression": "",
      "dimensions": {
        "InstanceId": "$instance"
      },
      "region": "$region",
      "id": "",
      "alias": "a",
      "statistics": [
        "Average"
      ],
      "period": "5m",
      "matchExact": true,
      "refId": "A"
}
```

when I navigate in explore all the other variables are interpolated except from the dimensions

Reply:
Good catch, thank you!

Should be good now :)